Sources/StreamChat/StateLayer/MessageState+Observer.swift

```diff
@@ -27,29 +27,31 @@ extension MessageState {
                     deletedMessagesVisibility: clientConfig.deletedMessagesVisibility,
                     shouldShowShadowedMessages: clientConfig.shouldShowShadowedMessages
                 ),
-                itemCreator: { try $0.asModel() as ChatMessage },
-                sorting: []
+                itemCreator: { try $0.asModel() as ChatMessage }
             )
         }
         
         struct Handlers {
-            let messageDidChange: (ChatMessage) async -> Void
-            let reactionsDidChange: ([ChatMessageReaction]) async -> Void
+            let messageDidChange: ((message: ChatMessage, changedReactions: [ChatMessageReaction]?)) async -> Void
             let repliesDidChange: (StreamCollection<ChatMessage>) async -> Void
         }
         
         func start(with handlers: Handlers) {
             do {
-                var lastReactions: Set<ChatMessageReaction>?
-                try messageObserver.startObserving(didChange: { message in
+                var lastSortedReactions: [ChatMessageReaction]?
+                try messageObserver.startObserving(onContextDidChange: { message in
                     guard let message else { return }
-                    let currentReactions = message.latestReactions
-                    if lastReactions != currentReactions {
-                        lastReactions = currentReactions
-                        let sortedReactions = currentReactions.sorted(by: { $0.updatedAt > $1.updatedAt })
-                        await handlers.reactionsDidChange(sortedReactions)
+                    let changedReactions: [ChatMessageReaction]?
+                    let currentReactions = message.latestReactions.sorted(by: MessageState.reactionsSorting)
```

Comment:
Reactions should also be sorted by the customers. But I assume we do this in the UI components?
This will also change a bit when we have reactions v2.

Reply:
For now I just preserved what we are doing in the `UserSearchController`